src/server/game/Instances/InstanceSaveMgr.cpp

```diff
@@ -136,7 +136,11 @@ bool InstanceSaveMgr::DeleteInstanceSaveIfNeeded(InstanceSave* save, bool skipMa
         // clear respawn times (if map is loaded do it just to be sure, if already unloaded it won't do it by itself)
         Map::DeleteRespawnTimesInDB(save->GetMapId(), save->GetInstanceId());
 
-        delete save;
+        if (deleteSave)
+        {
+            delete save;
+        }
+
         return true;
```

Comment:
should `return true` right after the delete. The return is on the wrong side of the brace and always returns true.